### PR TITLE
Change baseline path from ccsm to cesm, and update ESMF of derecho

### PR DESCRIPTION
**Pull request by fischer-ncar**
For derecho,
      BASELINE_ROOT was set to ccsm_baselines, it should be cesm_baselines.
      Also update ESMF library to ESMF8.6.0b03.

### CODE REVIEW

File: machines/config_machines.xml [hunk 1]
```diff
@@ -1370,12 +1370,12 @@ This allows using a different mpirun command to launch unit tests
...
       <modules DEBUG="TRUE">
 	<command name="load">parallelio/2.6.0-debug</command>
-	<command name="load">esmf/8.5.0b23-debug</command>
+	<command name="load">esmf/8.5.0</command>
       </modules>
 
       <modules DEBUG="FALSE">
 	<command name="load">parallelio/2.6.0</command>
-	<command name="load">esmf/8.5.0b23</command>
+	<command name="load">esmf/8.5.0</command>
```
Comment (jedwards4b) [inline machines/config_machines.xml:1378]: Please change to esmf/8.6.0b03

**Review by jedwards4b — CHANGES_REQUESTED** [review]

File: machines/config_machines.xml (hunk 1 above)
Comment (fischer-ncar) [reply machines/config_machines.xml]: Done.

File: machines/config_machines.xml [hunk 2]
```diff
@@ -1370,12 +1370,12 @@ This allows using a different mpirun command to launch unit tests
 
       <modules DEBUG="TRUE">
 	<command name="load">parallelio/2.6.0-debug</command>
-	<command name="load">esmf/8.5.0</command>
+	<command name="load">esmf/8.6.0b03</command>
```
Comment (jedwards4b) [inline machines/config_machines.xml:1373]: Sorry but this one should be esmf/8.6.0b03-debug

**Review by jedwards4b — APPROVED** [review]

**Comment by sjsprecious** [comment]
@fischer-ncar Can you install the corresponding ESMF version for the nvhpc compiler, too?

**Comment by jedwards4b** [comment]
@sjsprecious it's there, because of the sane way that modules are done on derecho you don't need a separate entry for each compiler.

**Comment by sjsprecious** [comment]
Thanks @jedwards4b .

**Pull request merged**